### PR TITLE
Change the logging strategy 

### PR DESCRIPTION
**Pull request by duckoder**
Work in progress

### CODE REVIEW

**Comment by empijei** [comment]
Thank for this

File: intercept/requests.go [hunk 1]
```diff
@@ -52,12 +52,12 @@ func handleRequest(preq *pendingRequest) {
...
 		if err != nil {
 			//TODO check this error and hijack connection to send raw bytes
-			log.Println("Error during provided response parsingh")
+			lg.Errorf("Error during provided response parsingh\n")
 		}
 		status.addRawEditedResponse(preq.id, editedRequestDump)
 	default:
 		//TODO implement this
-		log.Println("Not implemented yet")
+		lg.Errorf("Not implemented yet\n")
```
Comment (empijei) [inline intercept/requests.go]: You could use lg.Error() instead of Errorf with a \n at the end.

Comment (duckoder) [reply intercept/requests.go]: Yes, I'll change it, just missed it

**Comment by empijei** [comment]
I like what you are doing with the code, just please keep in mind there is a "Error" function so you don't have to use "Errorf" with a "\n" at the end. Such functions exist for every logging one. Available functions are:
```go
// Failure behaves as all other logging functions, then panics with the given parameters
func Failure(params ...interface{}) {}
// Failuref behaves as all other logging functions, then panics with the given parameters
func Failuref(format string, params ...interface{}){}
func Debug(params ...interface{})                 { _log(Level_Debug, params...) }
func Debugf(format string, params ...interface{}) { _logf(Level_Debug, format, params...) }
func Info(params ...interface{})                  { _log(Level_Info, params...) }
func Infof(format string, params ...interface{})  { _logf(Level_Info, format, params...) }
func Warn(params ...interface{})                  { _log(Level_Warn, params...) }
func Warnf(format string, params ...interface{})  { _logf(Level_Warn, format, params...) }
func Error(params ...interface{})                 { _log(Level_Error, params...) }
func Errorf(format string, params ...interface{}) { _logf(Level_Error, format, params...) }
```
Note: please remember that unused imports break the build, you can use [goimport](https://godoc.org/golang.org/x/tools/cmd/goimports) to automate the process of fixing imports

**Comment by duckoder** [comment]
@empijei can you help with the codebeat ? I don't understanding where is bug

File: intercept/history.go [hunk 2]
```diff
@@ -49,7 +49,7 @@ func (h *History) addResponse(ID int, res *http.Response) {
 	tmp, err := httputil.DumpResponse(res, true)
 	if err != nil {
 		//TODO
-		log.Println(err.Error())
+		lg.Failure(err.Error())
```
Comment (empijei) [inline intercept/history.go]: Should be lg.Error

File: intercept/history.go [hunk 3]
```diff
@@ -69,9 +69,9 @@ func StatusDump(status *History) {
 	status.RLock()
 	foo, err := json.MarshalIndent(status, " ", " ")
 	if err != nil {
-		log.Println(err.Error())
+		lg.Failure(err.Error())
```
Comment (empijei) [inline intercept/history.go]: should be lg.Error

File: intercept/responses.go [hunk 4]
```diff
@@ -52,7 +52,7 @@ func handleResponse(presp *pendingResponse) {
 		editedResponse = caseDrop()
 	default:
 		//TODO implement this
-		log.Println("Not implemented yet")
+		lg.Info("Not implemented yet")
```
Comment (empijei) [inline intercept/responses.go]: should be lg.Error

File: mitm/mitm.go [hunk 5]
```diff
@@ -180,14 +181,14 @@ func (p *Proxy) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 			TLSConfig: p.TLSServerConfig,
 		})
 		if err := sc.Handshake(); err != nil {
-			log.Println("Server Handshake:", err)
+			lg.Error("Server Handshake: %v", err)
```
Comment (empijei) [inline mitm/mitm.go]: should either be Errorf or Error without the "%v"

File: mitm/mitm.go [hunk 6]
```diff
@@ -180,14 +181,14 @@ func (p *Proxy) ServeHTTP(w http.ResponseWriter, req *http.Request) {
...
 			return
 		}
 	}
 
 	cc, err := p.tlsDial(req.Host, sc.ServerName)
 	if err != nil {
-		log.Println("tlsDial:", err)
+		lg.Error("tlsDial: %v", err)
```
Comment (empijei) [inline mitm/mitm.go]: should either be Errorf or Error without the "%v"

**Review by empijei — CHANGES_REQUESTED** [review]
I added some comments, after those fixes are implemented I will merge the PR. Thanks for your work!

**Comment by empijei** [comment]
@zhikiri ignore the codebeat metrics for now, I will fix it in time

**Comment by duckoder** [comment]
I'll fix those stupid mistakes shortly. 
(was interrupted to other projects)

**Comment by duckoder** [comment]
I've just forgot to commit... :)

**Comment by empijei** [comment]
Merged with https://github.com/empijei/wapty/commit/29434f5a880fe178f1e9589c096776f3ca1576d6 thanks for this :)

**Pull request closed without merging**